packages/gcf-utils/src/gcf-utils.ts

```diff
@@ -41,6 +48,44 @@ interface EnqueueTaskParams {
   name: string;
 }
 
+/**
+ * A pino-based singleton logger for use within Google Cloud Functions.
+ * Provides a metric() method to log metrics-related data
+ */
+export class GCFLogger {
+  private static logger: pino.Logger;
+
+  /**
+   * Get the logger instance
+   * @returns a pino-based logger
+   */
+  public static get(): pino.Logger {
+    if (!this.logger) {
+      this.logger = this.initLogger();
+    }
+    return this.logger;
+  }
+
+  private static initLogger(
+    options?: pino.LoggerOptions,
+    dest?: pino.DestinationStream
+  ): pino.Logger {
+    const defaultOptions: pino.LoggerOptions = {
+      customLevels: {
+        metric: 30,
+      },
+      level: 'trace',
+    };
+    if (options) {
+      Object.assign(defaultOptions, options);
```

Comment:
I think there's a typescript equivalent for this:
```
options = {...defaultOptions, ...options};
```
or something like that.

Reply:
cool, that looks more elegant. I'll take a look into that